### PR TITLE
drivers: npcx/nct38xx: drop DRV_CONFIG/DRV_DATA usage

### PR DESCRIPTION
**Pull request by WealianLiao**
Issue #37616 wants to make access to devise data and config consistent.
NPCX & NCT38XX use DRV_CONFIG/DRV_DATA macros to access config & data so npcx/nct38xx drivers don't modify at #41918.
This PR removes DRV_CONFIG/DRV_DATA to make access to devise data and config consistent.

### CODE REVIEW

File: drivers/interrupt_controller/intc_miwu.c [hunk 1]
```diff
@@ -158,42 +155,48 @@ static void intc_miwu_isr_pri(int wui_table, int wui_group)
 /* Platform specific MIWU functions */
 void npcx_miwu_irq_enable(const struct npcx_wui *wui)
 {
-	const uint32_t base = DRV_CONFIG(miwu_devs[wui->table])->base;
+	const uint32_t base =
+		((const struct intc_miwu_config *)miwu_devs[wui->table]->config)->base;
```
Comment (gmarull) [inline drivers/interrupt_controller/intc_miwu.c:159]: to improve readability, better do this:
```c
const struct intc_miwu_config *config = miwu_devs[wui->table]->config;
const uint32_t base = config->base;
```

**Review by gmarull — CHANGES_REQUESTED** [review]
Thanks for updating these drivers, please check my comments.

File: drivers/adc/adc_npcx.c [hunk 2]
```diff
@@ -93,8 +90,9 @@ static void adc_npcx_isr(void *arg)
 		/* Get result for each ADC selected channel */
 		while (data->channels) {
 			channel = find_lsb_set(data->channels) - 1;
-			result = GET_FIELD(CHNDAT(DRV_CONFIG((const struct device *)arg)->base,
-						  channel),
+			result = GET_FIELD(
+				CHNDAT(((const struct adc_npcx_config *)dev->config)->base,
```
Comment (gmarull) [inline drivers/adc/adc_npcx.c:94]: nit: for better readability, dereference config first

```c
const struct adc_npcx_config *config = dev->config;
...
result = GET_FIELD(CHNDAT(config->base, channel), NPCX_CHNDAT_CHDAT_FIELD);
```

**Review by keith-zephyr — APPROVED** [review]

File: drivers/interrupt_controller/intc_miwu.c (hunk 1 above)
Comment (WealianLiao) [reply drivers/interrupt_controller/intc_miwu.c]: done.

File: drivers/adc/adc_npcx.c (hunk 2 above)
Comment (WealianLiao) [reply drivers/adc/adc_npcx.c]: done.

**Review by MulinChao — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by WealianLiao** [comment]
Hi release managers, can this get merged?

**Pull request merged**